test/StandardAssetRegistry.js

```diff
@@ -73,4 +75,66 @@ contract('StandardAssetRegistry', accounts => {
     })
   })
 
+  describe('exists', () => {
+    it('ensures the asset exists after created', async () => {
+      const output = await registry.exists(1)
+      output.should.be.true      
+    })
+    it('ensures does return an assets if it does not exist', async () => {
+      const output = await registry.exists(100)
+      output.should.be.false
+    })
+    it('throws is not valid id', async () => {
+      return Promise.all([
+        registry.exists(true).should.be.rejected
+      ])
+    })
+    it('throws is not id is provided', async () => {
+      return Promise.all([
+        registry.exists().should.be.rejected
+      ])
+    })
+  })
+
+  describe('holderOf', () => {
+    it('should match the holder of the asset', async () => {
+      const one = '0xdf08f82de32b8d460adbe8d72043e3a7e25a3b39'
+      const two = '0x0000000000000000000000000000000000000000'
+      const outputOne = await registry.holderOf(1)
+      const outputTwo = await registry.holderOf(2)
+      outputOne.should.be.equal(one)
+      outputTwo.should.be.equal(two)
+    })
+    it('throws is not valid id', async () => {
+      return Promise.all([
+        registry.holderOf(true).should.be.rejected
+      ])
+    })
+    it('throws is not id is provided', async () => {
+      return Promise.all([
+        registry.holderOf().should.be.rejected
+      ])
+    })
+  })
+
+  describe('assetData', async () => {
+    it ('should returns the proper data', async () => {
+      const data = '0x6704fbfcd5ef766b287262fa2281c105d57246a6'
+      const output = await registry.assetData(1)
+      output.should.be.equal(data)
```

Comment:
Cool! first bug discovered. Where did that `0x67...` number came from? I think that's the address!

Reply:
return when call `registry.assetData(1)`

Reply:
The problem is that in the beforeEach function this happens:

```
  beforeEach(async function() {
    registry = await StandardAssetRegistry.new(creationParams)
    await registry.generate(0, user, sentByCreator)
    await registry.generate(1, user, sentByCreator)
  })
```

Generate is storing the user address as content which is a bit confusing. Maybe we can store some other constant metadata on those calls.

Reply:
Maybe even better have an `await registry.generate(0, 'test content', sentByCreator)` as a precondition to the specific test.

Reply:
The generate() function sends (assetId, content) as used in the beforeEach.
It's using `function generate(uint256 assetId, string data) public`

In this case data == user (addr) which is confusing.

Reply:
Let's make it clear by using a constant value for the data

`await registry.generate(0, CONTENT_DATA, user, sentByCreator)`